data/impl/RecordSet.js

```diff
@@ -63,20 +63,45 @@ export class RecordSet {
         if (!filter) return this;
 
         const passes = new Map(),
-            {records} = this;
-
-        // A record that passes the filter also recursively passes all its parents.
-        const markPass = (rec) => {
-            if (passes.has(rec.id)) return;
-            passes.set(rec.id, rec);
-            const {parent} = rec;
-            if (parent) markPass(parent);
-        };
-
-        records.forEach(rec => {
-            if (filter(rec)) markPass(rec);
+            isMarked = (rec) => passes.has(rec.id),
+            mark = (rec) => passes.set(rec.id, rec);
+
+        // Pass 1.  Mark all passing records, and potentially their children recursively.
+        // Any row already marked will already have all of its children marked, so check can be skipped
+        const includeChildren = withDefault(filter.includeChildren, false);
```

Comment:
Is the `withDefault` here necessary? `filter.includeChildren` would be falsey when omitted anyways, right?

Reply:
I prefer withDefault() to be more explicit, and symmetrical with `true` defaults...